src/server/firefox/ffPage.ts

```diff
@@ -90,15 +90,17 @@ export class FFPage implements PageDelegate {
       helper.addEventListener(this._session, 'Page.webSocketFrameReceived', this._onWebSocketFrameReceived.bind(this)),
       helper.addEventListener(this._session, 'Page.webSocketFrameSent', this._onWebSocketFrameSent.bind(this)),
     ];
-    this._pagePromise = new Promise(f => this._pageCallback = f);
+    this._pagePromise = Promise.all([
```

Comment:
This seems to introduce a race, where we will potentially miss some events (e.g. main network request in a popup), because we do not report the page until `Page.addScriptToEvaluateOnNewDocument` resolves, and we also do not pause the page.

Note that in other browsers this works because we manually resume the page after initialization.

Reply:
I checked this scenario when I was writing the patch, but things are hairy here so I might have missed something. Playwright internally operates on an uninitialized page. But I think you are right that the user might miss some events.